error-handling-core/src/main/java/com/bakdata/kafka/ErrorHeaderProcessor.java

```diff
@@ -136,7 +140,8 @@ public void process(final FixedKeyRecord<K, ProcessingError<V>> inputRecord) {
         addHeader(EXCEPTION_MESSAGE, value.getThrowable().getMessage(), headers);
         addHeader(EXCEPTION_STACK_TRACE, ExceptionUtils.getStackTrace(value.getThrowable()), headers);
         addHeader(DESCRIPTION, this.description, headers);
-        this.context.forward(inputRecord.withValue(value.getValue()));
+        addHeader(INPUT_TIMESTAMP, Long.toString(inputRecord.timestamp()), headers);
+        this.context.forward(inputRecord.withValue(value.getValue()).withTimestamp(this.context.currentSystemTimeMs()));
```

Comment:
I would argue that we should not modify timestamp here. The idea of using record headers is that you can reprocess erroneous records directly from DL topic. And timestamp might be relevant to the application. Therefore, I would leave timestamp unchanged and user needs to configure dead letter topic with the same config as input topic. Maybe add a header for current time.